R/aggregate_by_chi_zihao.R

```diff
@@ -0,0 +1,195 @@
+#' Aggregate by CHI
+#'
+#' @description Aggregate episode file by CHI to convert into
+#' individual file.
+#'
+#' @importFrom data.table .N
+#' @importFrom data.table .SD
+#'
+#' @inheritParams create_individual_file
+aggregate_by_chi_zihao <- function(episode_file) {
+  cli::cli_alert_info("Aggregate by CHI function started at {Sys.time()}")
+
+  episode_file <- episode_file %>%
+    dplyr::select(-c(postcode, gpprac)) %>%
+    dplyr::rename(
+      "gpprac" = "most_recent_gpprac",
+      "postcode" = "most_recent_postcode"
+    ) %>%
+    dplyr::select(-c(
+      dplyr::ends_with("_gpprac"),
+      dplyr::ends_with("_postcode"),
+      dplyr::ends_with("_DoB")
+    ))
+
+  data.table::setDT(episode_file) # Convert to data.table
+
+  # Sort the data within each chunk
+  data.table::setkeyv(
+    episode_file,
+    c(
+      "chi",
+      "record_keydate1",
+      "keytime1",
+      "record_keydate2",
+      "keytime2"
+    )
+  )
+
+  data.table::setnames(
+    episode_file,
+    c(
+      "ch_chi_cis", "cij_marker", "ooh_case_id"
+      # ,"hh_in_fy"
+    ),
+    c(
+      "ch_cis_episodes", "cij_total", "ooh_cases"
+      # ,"HL1_in_FY"
+    )
+  )
+
+  # Initialize an empty data.table for the aggregated results
+  aggregated_data <- data.table::data.table()
+
+  # Process the data in chunks
+  chunk_size <- min(nrow(episode_file), 5e7)
+  # Adjust the chunk size as per your system's memory capacity
+  n_chunks <- nrow(episode_file) %/% chunk_size
+
+
+  # colums specification
+  cols2 <- names(episode_file)[grepl("postcode$|DoB$|gpprac$",
+    names(episode_file),
+    ignore.case = TRUE
+  )]
```

Comment:
I think this is equivalent, `str_ends` is makes the regex more readable I think.
```suggestion
  cols2 <- names(episode_file)[stringr::str_ends(
  string = names(episode_file),
  pattern = stringr::regex(
    "postcode|DoB|gpprac",
    ignore_case = TRUE
  )
)]
```